src/widgets/MarkdownLabel.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "widgets/Label.hpp"
+
+#include <memory>
+
+class QTextDocument;
+
+namespace chatterino {
+
+/// @brief A Label that supports rendering markdown text
+///
+/// MarkdownLabel inherits from Label and adds markdown rendering capabilities.
+/// It automatically handles markdown document creation and rendering.
+class MarkdownLabel : public Label
+{
+public:
+    explicit MarkdownLabel(QString text = QString(),
+                           FontStyle style = FontStyle::UiMedium);
+    explicit MarkdownLabel(BaseWidget *parent, QString text = QString(),
+                           FontStyle style = FontStyle::UiMedium);
```

Comment:
Seems like we only use the default-constructor. I'd prefer if you only create one constructor that requires the parent & default text, meaning the places where you create this with the LayoutCreator emplace would need to change a bit.

so something like
```c++
MarkdownLabel(BaseWidget *parent, QString text, FontStyle style = ....)
```